### PR TITLE
Add alt visitor login

### PR DESCRIPTION
**Pull request by eliascodes**
Related to #536

#### Changes
* Rename `main` to something meaningful
* Add alternate sign-up option for visitors

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { Route, Switch } from 'react-router-dom';
 
 import redirectAfterTimeout from './shared/components/hoc/redirect_after_timeout';
 
-import Main from './visitors/pages/main';
+import VisitorSignupPage from './visitors/pages/Signup';
```
Comment (astroash) [inline src/App.js]: 🙇🏽‍♀️

File: src/util.js [hunk 2]
```diff
@@ -61,3 +63,37 @@ export const toCancellable = (p) => {
...
   return wrappedPromise;
 };
+
+/**
+ * Utility function to support client-side redirects based on
+ * common status codes
+ *
+ * Allows passing custom redirects to override defaults
+ */
+export const redirectOnError = (to, error, custom = {}) => {
```
Comment (astroash) [inline src/util.js]: Since we've got this, can it be implemented on all requests?

File: src/App.js (hunk 1 above)
Comment (eliascodes) [reply src/App.js]: I don't know what this emoji means.

**Comment by astroash** [comment]
@eliascodes can you verbally review my changed and merge

File: src/cb_admin/pages/Dashboard.js [hunk 3]
```diff
@@ -51,23 +52,7 @@ export default class Dashboard extends React.Component {
...
-        } else if (ErrorUtils.errorStatusEquals(error, 404)) {
-          this.props.history.push('/error/404');
-
-        } else {
-          this.props.history.push('/error/unknown');
-
-        }
-      });
-
+      .catch(err => redirectOnError(this.props.history.push, err, { 401: '/admin/login' }));
```
Comment (eliascodes) [inline src/cb_admin/pages/Dashboard.js]: `/admin/login` won't work, they need to log in again.

File: src/cb_admin/pages/ForgotPassword.js [hunk 4]
```diff
@@ -35,22 +36,15 @@ export default class ForgotPassword extends React.Component {
...
-        } else if (ErrorUtils.errorStatusEquals(error, 401)) {
-          this.setState({ errors: { email: error.response.data.error } });
-
-        } else if (ErrorUtils.errorStatusEquals(error, 500)) {
-          this.props.history.push('/error/500');
-
-        } else if (ErrorUtils.errorStatusEquals(error, 404)) {
-          this.props.history.push('/error/404');
+        } else if (ErrorUtils.errorStatusEquals(err, 401)) {
+          this.setState({ errors: { email: err.response.data.error } });
```
Comment (eliascodes) [inline src/cb_admin/pages/ForgotPassword.js]: Is this right? In `Login.js` it's `err.response.data.error.message`.

File: src/cb_admin/pages/Settings.js [hunk 5]
```diff
@@ -119,15 +119,7 @@ export default class SettingsPage extends React.Component {
...
-      .catch((error) => {
-        if (ErrorUtils.errorStatusEquals(error, 500)) {
-          this.props.history.push('/error/500');
-        } else if (ErrorUtils.errorStatusEquals(error, 401)) {
-          this.props.history.push('/admin/login');
-        } else {
-          this.props.history.push('/error/unknown');
-        }
-      });
+      .catch(err => redirectOnError(this.props.history.push, err, { 401: '/admin/login' }));
```
Comment (eliascodes) [inline src/cb_admin/pages/Settings.js]: See other comment on `/admin/login`

File: src/cb_admin/pages/Visitor.js [hunk 6]
```diff
@@ -121,42 +121,17 @@ export default class VisitorProfile extends React.Component {
...
-          this.props.history.push('/admin/login');
-        } else if (ErrorUtils.errorStatusEquals(error, 500)) {
-          this.props.history.push('/error/500');
-        } else if (ErrorUtils.errorStatusEquals(error, 404)) {
-          this.props.history.push('/error/404');
-        } else {
-          this.props.history.push('/error/unknown');
-        }
-      });
+      .catch(err => redirectOnError(this.props.history.push, err, { 401: '/admin/login' }));
```
Comment (eliascodes) [inline src/cb_admin/pages/Visitor.js]: Same comment on `/admin/login`

File: src/cb_admin/pages/ForgotPassword.js (hunk 4 above)
Comment (astroash) [reply src/cb_admin/pages/ForgotPassword.js]: good spot, error response was mocked incorrectly in the tests too.

Fixed in amend.

File: src/cb_admin/pages/Dashboard.js (hunk 3 above)
Comment (astroash) [reply src/cb_admin/pages/Dashboard.js]: Fixed in amend.

File: src/cb_admin/pages/Settings.js (hunk 5 above)
Comment (astroash) [reply src/cb_admin/pages/Settings.js]: Fixed in amend.

File: src/cb_admin/pages/Visitor.js (hunk 6 above)
Comment (astroash) [reply src/cb_admin/pages/Visitor.js]: Fixed in amend.

**Pull request merged**